### PR TITLE
Homework 3

### PR DESCRIPTION
**Pull request by IsaevMH**

### CODE REVIEW

File: src/test/java/com/epam/tc/hw3/page/IndexPage.java [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+    private WebElement frameButton;
+    @FindBy(xpath = "//ul[@class='sidebar-menu left']/li")
+    private List<WebElement> leftMenuOptions;
+    @FindBy(xpath = "//div//ul/li[@class='dropdown']")
+    private WebElement serviceButton;
+    @FindBy(xpath = "//ul/li/a[text() ='Different elements']")
+    private WebElement differentElementsOption;
+
+    public IndexPage(WebDriver driver) {
+        super(driver);
```
Comment (ZeroShadows) [inline src/test/java/com/epam/tc/hw3/page/IndexPage.java:45]: Refactor these tests in Void Page Object design pattern.

File: src/test/java/com/epam/tc/hw3/page/IndexPage.java [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+import java.util.List;
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.FindBy;
+import org.openqa.selenium.support.PageFactory;
+
+public class IndexPage extends AbstractPage {
+
+    public static String BASE_URL = "https://jdi-testing.github.io/jdi-light/index.html";
```
Comment (ZeroShadows) [inline src/test/java/com/epam/tc/hw3/page/IndexPage.java:13]: u have this Base_url into BaseTest and u have it here.
lets remove it from here

File: src/test/java/com/epam/tc/hw3/test/BaseTest.java [hunk 3]
```diff
@@ -0,0 +1,25 @@
+package com.epam.tc.hw3.test;
+
+import com.epam.tc.hw3.driver.Driver;
+import com.epam.tc.hw3.service.DataReader;
+import org.openqa.selenium.WebDriver;
+import org.testng.annotations.*;
+import org.testng.asserts.SoftAssert;
+
+public class BaseTest {
+    protected static final String BASE_URL = "https://jdi-testing.github.io/jdi-light/index.html";
```
Comment (ZeroShadows) [inline src/test/java/com/epam/tc/hw3/test/BaseTest.java:10]: can we move that into properties section?

File: src/test/java/com/epam/tc/hw3/page/DifferentElementsPage.java [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+import java.util.ArrayList;
+import java.util.List;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.FindBy;
+import org.openqa.selenium.support.PageFactory;
+
+public class DifferentElementsPage extends AbstractPage {
+
+    private static String BASE_URL = "https://jdi-testing.github.io/jdi-light/different-elements.html";
```
[the remark this answers is not in the document]
Comment (IsaevMH) [reply src/test/java/com/epam/tc/hw3/page/DifferentElementsPage.java:13]: Issue is fixed

File: src/test/java/com/epam/tc/hw3/page/IndexPage.java (hunk 2 above)
Comment (IsaevMH) [reply src/test/java/com/epam/tc/hw3/page/IndexPage.java:13]: Issue is fixed

File: src/test/java/com/epam/tc/hw3/test/BaseTest.java (hunk 3 above)
Comment (IsaevMH) [reply src/test/java/com/epam/tc/hw3/test/BaseTest.java:10]: Issue is fixed

File: src/test/java/com/epam/tc/hw3/page/DifferentElementsPage.java [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+
+    public DifferentElementsPage(WebDriver driver) {
+        super(driver);
+        PageFactory.initElements(driver, this);
+    }
+
+    @Override
+    public DifferentElementsPage openPage() {
+        driver.navigate().to(BASE_URL);
+        return this;
```
[the remark this answers is not in the document]
Comment (IsaevMH) [reply src/test/java/com/epam/tc/hw3/page/DifferentElementsPage.java:34]: Issue is fixed

File: src/test/java/com/epam/tc/hw3/page/IndexPage.java (hunk 1 above)
Comment (IsaevMH) [reply src/test/java/com/epam/tc/hw3/page/IndexPage.java:45]: Issue is fixed

**Pull request merged**